plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java

```diff
@@ -87,13 +96,24 @@ public static FetchSourceContext getSourceContext(RestRequest request, SearchSou
         } else {
             // When user does not set the _source field in search model api request, searchSourceBuilder.fetchSource becomes null
             if (!userAgent.contains(OPENSEARCH_DASHBOARDS_USER_AGENT)) {
-                return new FetchSourceContext(true, Strings.EMPTY_ARRAY, UI_METADATA_EXCLUDE);
+                return new FetchSourceContext(true, Strings.EMPTY_ARRAY, ArrayUtils.add(UI_METADATA_EXCLUDE, "model_content"));
             } else {
-                return null;
+                return new FetchSourceContext(true, Strings.EMPTY_ARRAY, new String[] { "model_content" });
             }
         }
     }
 
+    /**
+     * Return FetchSourceContext
+     * @param returnModelContent if the model content should be returned
+     */
+    public static FetchSourceContext getFetchSourceContext(boolean returnModelContent) {
+        if (!returnModelContent) {
+            return new FetchSourceContext(true, Strings.EMPTY_ARRAY, new String[] { "content", "model_content" });
```

Comment:
How about using `MLModel.OLD_MODEL_CONTENT_FIELD` and `MLModel.MODEL_CONTENT_FIELD`?